### ISSUE TITLE
MPQs opened with readonly

### ISSUE THREAD

**Issue by stijnherfst** [ladislav-zezula/StormLib#125]
Hi, when using stormlib (or Ladik's MPQ Editor) the [following map file](https://stack.stijnv.nl/s/xyTiNIT7Pv2c5eX) is opened in readonly mode. I'm suspecting MPQ_FILE_FIX_KEY, but I'm not sure.

**Comment by ladislav-zezula** [comment]
Most of the time, it's due the map being malformed in order to prevent MPQ editors from reading files.

**Comment by flowtsohg** [comment]
That does not prevent you from writing them, since all of these issues are automatically fixed upon saving.
Is this to support so called map protectors?

**Comment by ladislav-zezula** [comment]
Yes. The amount, methods and complexity of map protectors led me to a decision to open protected maps for read-only access. You can allways scan names, extract as much as possible and create a new map.

**Comment by flowtsohg** [comment]
Exactly, so what is the point? Seems like a counter intuitive decision for an open source project, but if that's how you see it, that's how it is.

**Comment by Unryze** [comment]
First of all, it is very hard to make MPQ Editor to write the data without breaking maps with very strong corruptions or for example a JAPI "protection" that is applied with YDWE.

Second of all, MPQ Editor is not made around WC3 and he should not do so, nor try to force MPQ Editor to suffer in other areas, because of WC3.

Third of all, it a map is read-only and you want to edit it, that means you are trying to break the protection, which yet again is not something important, nor even good for an open source project as it "assists theft".

Fourth of all, this serves as a setback to majority of users that only care about stealing others work and will never ever do anything else, I for example UNHEXED every single type of protection, before submitting them to Ladik, meaning if you want to remove read-only you can do it with HxD (HexEditor).

TL;DR: Do no make Ladik do more, than he already did, and if you really want to deprotect maps, then learn hex editing.

**Comment by flowtsohg** [comment]
I have my own implementation that writes anything it can, I don't have any issue. I just legitimately never understood this support for obfuscating data in the wc3 modding community, this is not what modding should be about.

**Comment by Unryze** [comment]
@flowtsohg can you name a map with a strong corruption that you can edit no problem? I am interested in that.

Also, Warcraft 3 modding has nothing to do with MPQ now does it? Yes, you can modify .mpq / .mix, but that is not really modding if you are deprotecting / editing someone else's files, I do agree with you partially though, but as I said, MPQ editor is no made around WC3 and I really don't want it to be.

**Comment by flowtsohg** [comment]
I probably don't know about every protection (care to share maps? :P) but I can open boba/spaz no issues (at least those are the names someone gave me for those cute offset changes etc.)

**Comment by Unryze** [comment]
@flowtsohg boba should be somj2hm16, there is also kangtoolee a modified version of it, then JAPI (YDWE "corruption" ), then there is very old and useless Moonlight, if you want an example, well here you go:

Map with JAPI should be here: https://github.com/ladislav-zezula/StormLib/issues/117 you can read some info that I posted.

Or this: https://github.com/ladislav-zezula/StormLib/issues/105 EdenRPG with tons of hex protection, but it is openable atm with Read-Only, same as every other map I reported.

And lets end with this: https://github.com/ladislav-zezula/StormLib/issues/81 (I hope the file is still there).

**Comment by flowtsohg** [comment]
Thanks I'll take a look!

**Comment by flowtsohg** [comment]
I definitely do not support these.
Not that this is related to the maps we tested and have no reason to be in read-only.
Third map isn't available.

Thanks :)

**Comment by Unryze** [comment]
@flowtsohg Well, the thing is, your map may have a little corruption, but others have it in a different way, you can't make MPQ Editor differentiate between them and add to the corrupted file, the only way is, to make it rebuild the MPQ if it is read-only, but that won't be any good, if you don't have full listfile.

Either way, I am just trying to make you understand, that there are too many types of protections and you can't force MPQ Editor to support and understand them all.

**Comment by flowtsohg** [comment]
Yes, I mean specifically the ones we have tested - I didn't understand why they are read-only when my implementation can write them.
Regardless, it's fine.

**Comment by ladislav-zezula** [comment]
Thanks for your opinions guys. Like some said, it's "MPQ Editor", not "WC3 Deprotector". And it will stay as it is. I will add support for new MPQ protections, but will keep opening them in read-only mode.
